service/history/replication/dlq_handler.go

```diff
@@ -169,21 +176,14 @@ func (r *dlqHandlerImpl) fetchAndEmitMessageCount(ctx context.Context) error {
 }
 
 func (r *dlqHandlerImpl) emitDLQSizeMetricsLoop() {
-	getInterval := func() time.Duration {
-		return backoff.JitDuration(
-			dlqMetricsEmitTimerInterval,
-			dlqMetricsEmitTimerCoefficient,
-		)
-	}
-
-	timer := time.NewTimer(getInterval())
+	timer := time.NewTimer(r.getInterval())
```

Comment:
I recommend using clock.TimeSource to mimic time-related behaviour.

It allows to have the same behavior for prod and be able to mimic time changes in tests without any time delays.

Reply:
Good point. I've made and attempt to fix that. Unfortunately the `timer` was set inside of the method being called by the goroutine, so I had to make further changes in order to be able to use `Advance` without using `sleep`. Let me know what you think.